### PR TITLE
Removing the packager from storybook.

### PR DESCRIPTION
**Pull request by Gongreg**
Issue: #4179

By having packager logic inside storybook we are prone for breaking things.
We have to support metro, haul, typescript and other cases. All of these things are responsibility of the user.

To avoid it I've completely removed packager logic from storybook.

What is more starting from v5.0 the default behaviour will be to skip the server altogether and include StorybookUI in your app so this is a first step to that direction.

I've added some info how to bring back old behaviour @petrbela suggested it.

### CODE REVIEW

**Comment by igor-dv** [comment]
I think it's rather an important breaking change, so having a note in the MIGRATION.md will be a good thing

**Comment by Gongreg** [comment]
@igor-dv Definitely.

File: app/react-native/src/bin/storybook-start.js [hunk 1]
```diff
@@ -48,61 +39,3 @@ server.listen(...listenAddr, err => {
...
-  let roots = [projectDir];
-
-  if (program.root) {
-    roots = roots.concat(program.root.split(',').map(root => path.resolve(root)));
-  }
-
-  try {
-    // eslint-disable-next-line global-require
-    require('babel-register')({
-      presets: [require.resolve('babel-preset-flow')],
```
Comment (igor-dv) [inline app/react-native/src/bin/storybook-start.js]: do `babel-register` and `babel-preset-flow` still need to be installed?

Comment (Gongreg) [reply app/react-native/src/bin/storybook-start.js]: Good question. What were they used for in the first place?

Comment (igor-dv) [reply app/react-native/src/bin/storybook-start.js]: I have no idea =)

**Review by igor-dv — APPROVED** [review]

File: app/react-native/src/bin/storybook-start.js (hunk 1 above)
Comment (Gongreg) [reply app/react-native/src/bin/storybook-start.js]: Ah but it is flow and some other things. I will remove it. 
Thanks @igor-dv .

**Comment by igor-dv** [comment]
@Gongreg, is anything missing here ?

**Comment by Gongreg** [comment]
Nothing that I am aware of. We will do a blog post regarding this update. But it is not part of the PR

**Comment by jqn** [comment]
@igor-dv one thing that is missing from this is some instructions on how to switch from the storybook UI to the react native UI. The old behavior kept this separate but I don't see how this makes sense 
`module.exports = __DEV__ ? StorybookUI : App;` if I want to switch back and forth. there has to be a better way to switch. Also if I'm missing something here please let me know.

**Comment by Gongreg** [comment]
@jqn You could have a storybook tab in bottom nav menu. Or inside admin panel. Or even the same switch you just wrote about. It shouldnt be more work to switch this way than killing packager and starting the new one. Or if you want to you can still start the packager old way, you just have to do it yourself. We will release a blogpost/update storybook website with information when 4.0 is released. Or you can check docs folder in @storybook/react-native code. (I am on mobile so it is hard to link it)

**Comment by kdawgwilk** [comment]
Or you can start the RN packager using the storybook dir: `react-native start --projectRoot storybook`

**Comment by kdawgwilk** [comment]
Actually I just tried to give this a go and with RN >= 0.56 this throws an error `error: unknown option `--projectRoot'`

**Comment by Gongreg** [comment]
It may be `--projectRoots`. From 0.57 it is `--projectRoot`. That is exactly the reason why we removed the packager. We don't want to maintain it.

**Comment by kdawgwilk** [comment]
It is `--projectRoots` but also giving a relative path doesn't work as expected and instead I need to give it a absolute path e.g. 
```
--projectRoots `pwd`/storybook
```

**Comment by esbenvb** [comment]
@kdawgwilk @Gongreg keep in mind that `--projectRoot` is not implemented in 0.57, but only documented. 0.57.1 fixes that
https://github.com/facebook/react-native/commit/fb109e9120e4366e40d7468625fcd6b14a5fe1e9

**Pull request merged**